### PR TITLE
React SDK

### PR DESCRIPTION
**Pull request by rrpff**

### CODE REVIEW

File: packages/react-sdk/src/index.tsx [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+  flags: FeatureFlagsOptions;
+  children?: ReactNode;
+  sdk?: BucketInstance;
+};
+
+export default function Bucket({ children, sdk, ...config }: BucketProps) {
+  const [flags, setFlags] = useState<Flags>({});
+  const [bucket] = useState(() => sdk ?? BucketSingleton);
+
+  const contextKey = JSON.stringify(config);
```
Comment (roncohen) [inline packages/react-sdk/src/index.tsx:38]: is this guaranteed to return something stable? We've used json-canonical in the past.

File: README.md [hunk 2]
```diff
@@ -12,6 +12,12 @@ Universal JS SDK for the browser and Node.js
 
 [Read the docs](packages/tracking-sdk/README.md)
 
+### React SDK
+
+Client side React SDK
+
+[Read the docs](packages/react-sdk/README.md)
```
Comment (roncohen) [inline README.md:19]: Let's add this README. We should have usage examples etc.

File: packages/react-sdk/src/index.tsx [hunk 3]
```diff
@@ -0,0 +1,68 @@
...
+
+  return <Context.Provider children={children} value={context} />;
+}
+
+export function useBucket() {
+  return useContext<BucketContext>(Context).bucket;
+}
+
+export function useFeatureFlags() {
+  return useContext<BucketContext>(Context).flags;
```
Comment (roncohen) [inline packages/react-sdk/src/index.tsx:63]: there's no wait to know if the flags are loading, right?

**Review by roncohen — COMMENTED** [review]
Wrote some comments from the initial quick look.

File: packages/react-sdk/src/index.tsx (hunk 3 above)
Comment (rrpff) [reply packages/react-sdk/src/index.tsx]: Have added loading states now

File: README.md (hunk 2 above)
Comment (rrpff) [reply README.md:19]: Done 👍 Review on the content appreciated

File: packages/react-sdk/src/index.tsx (hunk 1 above)
Comment (rrpff) [reply packages/react-sdk/src/index.tsx]: Good shout, have used that

File: packages/react-sdk/README.md [hunk 4]
```diff
@@ -0,0 +1,91 @@
...
+
+All options which can be passed to `bucket.init` can be passed as props to the Bucket higher order component.
+
+```tsx
+<Bucket
+  publishableKey="{YOUR_PUBLISHABLE_KEY}" // The publishable key of your app environment
+  debug={false} // Enable debug mode to log info and errors
+  persistUser={true} // See the Tracking SDK documentation under "Persisting Users"
+  host="https://tracking.bucket.co" // Configure the host Bucket calls are made to
+  sseHost="https://livemessaging.bucket.co" // Configure the host Bucket sse calls are made to
```
Comment (roncohen) [inline packages/react-sdk/README.md:33]:
```suggestion
  sseHost="https://livemessaging.bucket.co" // Configure the host Bucket SSE calls are made to
```

File: packages/react-sdk/README.md [hunk 5]
```diff
@@ -0,0 +1,91 @@
...
+
+This will initialize Bucket, fetch feature flags and start listening for Live Satisfaction events.
+
+```tsx
+<Bucket publishableKey="{YOUR_PUBLISHABLE_KEY}">{/* ... */}</Bucket>
+```
+
+### Props
+
+All options which can be passed to `bucket.init` can be passed as props to the Bucket higher order component.
```
Comment (roncohen) [inline packages/react-sdk/README.md:25]: probably worth adding a link to the readme there. Don't want to repeat everything here, for example LiveSatisfaction configuration.

File: packages/react-sdk/README.md [hunk 6]
```diff
@@ -0,0 +1,91 @@
...
+  persistUser={true} // See the Tracking SDK documentation under "Persisting Users"
+  host="https://tracking.bucket.co" // Configure the host Bucket calls are made to
+  sseHost="https://livemessaging.bucket.co" // Configure the host Bucket sse calls are made to
+  feedback={{
+    // See feedback options here: https://github.com/bucketco/bucket-tracking-sdk/blob/main/packages/tracking-sdk/FEEDBACK.md#global-feedback-configuration
+  }}
+>
+```
+
+## Hooks
```
Comment (roncohen) [inline packages/react-sdk/README.md:40]: hey @Munter @Swiftwork . Would be great to get your input on this API. I know you had some thoughts.

**Review by roncohen — APPROVED** [review]
LGTM. Let's get a pre-release out.

**Pull request merged**